plugins/fontdownloader.koplugin/main.lua

```diff
@@ -0,0 +1,527 @@
+        ["bengali"]             = _("Bengali"),
+        ["chinese-hongkong"]    = _("Hong Kong Chinese"),
+        ["chinese-simplified"]  = _("Simplified Chinese"),
+        ["chinese-traditional"] = _("Traditional Chinese"),
+        ["cyrillic"]            = _("Cyrillic"),
+        ["devanagari"]          = _("Devanagari"),
+        ["greek"]               = _("Greek"),
+        ["gujarati"]            = _("Gujarati"),
+        ["gurmukhi"]            = _("Gurmukhi"),
+        ["hebrew"]              = _("Hebrew"),
+        ["japanese"]            = _("Japanese"),
+        ["kannada"]             = _("Kannada"),
+        ["khmer"]               = _("Khmer"),
+        ["korean"]              = _("Korean"),
+        ["latin"]               = _("Latin"),
+        ["malayalam"]           = _("Malayalam"),
+        ["myanmar"]             = _("Myanmar"),
+        ["oriya"]               = _("Oriya"),
+        ["sinhala"]             = _("Sinhala"),
+        ["tamil"]               = _("Tamil"),
+        ["telugu"]              = _("Telugu"),
+        ["thai"]                = _("Thai"),
+        ["tibetan"]             = _("Tibetan"),
+        ["vietnamese"]          = _("Vietnamese"),
+    },
+    variants = {
+        ["regular"]             = "regular",
+        ["italic"]              = "italic",
+        ["500"]                 = "medium",
+        ["500italic"]           = "medium-italic",
+        ["700"]                 = "bold",
+        ["700italic"]           = "bold-italic",
+    },
+}
+
+function FontDownloader:init()
+    self.ui.menu:registerToMainMenu(self)
+    local keyfile = io.open(self.user_key, "r")
+    if keyfile then
+        local key = keyfile:read("*a")
+        keyfile:close()
+        self.api_key = key
+        return
+    end
+    self.api_key = "AIzaSyDQZaihK8Lb7jJ3DYyrQrpyyJF7tyvrqAs"
+end
+
+function FontDownloader:addToMainMenu(menu_items)
+    -- plugin settings
+    menu_items.font_downloader = {
+        text = _("Font downloader"),
+        sub_item_table = {
+            {
+                text = _("Set download location"),
+                callback = function()
+                    self:setFontDir()
+                end,
+            },
+            {
+                text = _("Sync font index"),
+                callback = function()
+                    self.fonts = self:fontTable()
+                end,
+            },
+        }
+    }
+    -- inject in font menu for CRE documents
+    if self.ui.document and self.ui.document.provider == "crengine" then
+        local menu_orig = menu_items.change_font.sub_item_table[1].sub_item_table
+        local position = #menu_orig - 1 -- above fallback fonts
+        local menu = menu_orig
+        table.insert(menu, position, {
+            text = _("Download more fonts"),
+            callback = function()
+                self:frontpage()
+            end,
+        })
+        menu_items.change_font.sub_item_table[1].sub_item_table = menu
```

Comment:
Here, `menu_items.change_font.sub_item_table[1].sub_item_table`, `menu_orig` and `menu` are all the same object - so no need for this kind of swap :)
```diff
     -- inject in font menu for CRE documents
     if self.ui.document and self.ui.document.provider == "crengine" then
-        local menu_orig = menu_items.change_font.sub_item_table[1].sub_item_table
-        local position = #menu_orig - 1 -- above fallback fonts
-        local menu = menu_orig
+        local menu = menu_items.change_font.sub_item_table[1].sub_item_table
+        local position = #menu - 1 -- above fallback fonts
         table.insert(menu, position, {
             text = _("Download more fonts"),
             callback = function()
                 self:frontpage()
             end,
         })
-        menu_items.change_font.sub_item_table[1].sub_item_table = menu
     end
```